lib/Doctrine/DBAL/Platforms/SqlitePlatform.php

```diff
@@ -642,6 +642,7 @@ protected function initializeDoctrineTypeMappings()
             'decimal'          => 'decimal',
             'numeric'          => 'decimal',
             'blob'             => 'blob',
+            'binary'           => 'blob',
```

Comment:
There may be a confusion between binary and blob types in the DBAL. The "binary" is supposed to represent a binary string of a known (fixed or variable) length like `BINARY(N)` or `VARBINARY(N)` in [MySQL](https://dev.mysql.com/doc/refman/5.7/en/binary-varbinary.html). Why "blob" is a large chunk of binary data.

The binary fields should be possible to use for comparison and as a primary key (e.g. a UUID PK could be stored in a `BINARY(16)`) field (see https://github.com/doctrine/dbal/issues/2787 for a reference).

Please make sure the way how "binary" is represented in SQLite meets these criteria.

Reply:
@spyric sorry for the confusion. According to the `BinaryTest` added as part of #3133, `BLOB` indeed can be used as implementation of the `BINARY` column type on SQLite.

Please provide a test case.